// angular-split/esm5/lib/interface.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @record
 */
export function IPoint() { }
if (false) {
    /** @type {?} */
    IPoint.prototype.x;
    /** @type {?} */
    IPoint.prototype.y;
}
/**
 * @record
 */
export function IArea() { }
if (false) {
    /** @type {?} */
    IArea.prototype.component;
    /** @type {?} */
    IArea.prototype.order;
    /** @type {?} */
    IArea.prototype.size;
    /** @type {?} */
    IArea.prototype.minSize;
    /** @type {?} */
    IArea.prototype.maxSize;
}
/**
 * @record
 */
export function ISplitSnapshot() { }
if (false) {
    /** @type {?} */
    ISplitSnapshot.prototype.gutterNum;
    /** @type {?} */
    ISplitSnapshot.prototype.allAreasSizePixel;
    /** @type {?} */
    ISplitSnapshot.prototype.allInvolvedAreasSizePercent;
    /** @type {?} */
    ISplitSnapshot.prototype.lastSteppedOffset;
    /** @type {?} */
    ISplitSnapshot.prototype.areasBeforeGutter;
    /** @type {?} */
    ISplitSnapshot.prototype.areasAfterGutter;
}
/**
 * @record
 */
export function IAreaSnapshot() { }
if (false) {
    /** @type {?} */
    IAreaSnapshot.prototype.area;
    /** @type {?} */
    IAreaSnapshot.prototype.sizePixelAtStart;
    /** @type {?} */
    IAreaSnapshot.prototype.sizePercentAtStart;
}
/**
 * @record
 */
export function ISplitSideAbsorptionCapacity() { }
if (false) {
    /** @type {?} */
    ISplitSideAbsorptionCapacity.prototype.remain;
    /** @type {?} */
    ISplitSideAbsorptionCapacity.prototype.list;
}
/**
 * @record
 */
export function IAreaAbsorptionCapacity() { }
if (false) {
    /** @type {?} */
    IAreaAbsorptionCapacity.prototype.areaSnapshot;
    /** @type {?} */
    IAreaAbsorptionCapacity.prototype.pixelAbsorb;
    /** @type {?} */
    IAreaAbsorptionCapacity.prototype.percentAfterAbsorption;
    /** @type {?} */
    IAreaAbsorptionCapacity.prototype.pixelRemain;
}
/**
 * @record
 */
export function IOutputData() { }
if (false) {
    /** @type {?} */
    IOutputData.prototype.gutterNum;
    /** @type {?} */
    IOutputData.prototype.sizes;
}
/**
 * @record
 */
export function IOutputAreaSizes() { }
